test/test_nvfuser_prims.py

```diff
@@ -0,0 +1,196 @@
+                if is_iterable_of_tensors(inp):
+                    tensor_list = []
+                    for x in inp:
+                        if isinstance(x, torch.Tensor) and x.requires_grad:
+                            tensor_list.append(inputs[input_idx])
+                            input_idx = input_idx + 1
+                        else:
+                            tensor_list.append(x)
+                    return tensor_list, input_idx
+                elif isinstance(inp, torch.Tensor) and inp.requires_grad:
+                    return inputs[input_idx], input_idx + 1
+                else:
+                    return inp, input_idx
+
+            def fn(*inputs):
+                # Puts inputs back into sample properly
+                positional_args = []
+                input_idx = 0
+                inp, input_idx = _input_recomposition_helper(
+                    inputs, sample.input, input_idx
+                )
+                positional_args.append(inp)
+
+                for x in sample.args:
+                    inp, input_idx = _input_recomposition_helper(inputs, x, input_idx)
+                    positional_args.append(inp)
+
+                # Recreates kwargs
+                kwargs = {}
+                for k, v in sample.kwargs.items():
+                    inp, input_idx = _input_recomposition_helper(inputs, v, input_idx)
+                    kwargs[k] = inp
+
+                output = op.gradcheck_wrapper(variant, *positional_args, **kwargs)
+                if sample.output_process_fn_grad is not None:
+                    return sample.output_process_fn_grad(output)
+                return output
+
+            self.assertTrue(
+                gradcheck(
+                    fn,
+                    gradcheck_args,
+                    check_grad_dtypes=True,
+                    nondet_tol=op.gradcheck_nondet_tol,
+                    fast_mode=op.gradcheck_fast_mode,
+                    check_undefined_grad=True,
+                    check_batched_grad=False,
+                )
+            )
+
+    @onlyCUDA
+    @skipCUDAIfRocm
+    @ops(ref_nvprims_ops, dtypes=OpDTypes.supported, allowed_dtypes=(torch.float64,))
+    def test_nvprims_grad(self, device, dtype, op):
+        from copy import copy
+
+        op = copy(op)
+        # Here we are testing aten executor because we are verifying that the
+        # backward pass is correct. gradcheck doesn't work with nvfuser executor
+        # because nvfuser's result is a detached tensor.
+        op.op = partial(make_traced(op.op), executor="aten")
+
+        self._check_helper(device, dtype, op, op.get_op())
+
+    @onlyCUDA
+    @skipCUDAIfRocm
+    @ops(ref_nvprims_ops, dtypes=OpDTypes.supported, allowed_dtypes=(torch.float64,))
+    def test_nvprims_grad_trace_backward(self, device, dtype, op):
+        # Check if result is boolean
+        if op.torch_opinfo_name in ["eq", "ne", "gt", "ge", "lt", "le", "isfinite"]:
```

Comment:
Instead of checking names use skips on the OpInfos, use the OpInfo metadata for whether the op supports autograd or not, or directly test whether the output is not a differentiable datatype